src/CSV.jl

```diff
@@ -561,89 +577,96 @@ function parsetape(::Val{transpose}, ignoreemptylines, ncols, typemap, tapes, ta
                     end
                 end
             end
-            tapeidx += 2
             pos > len && break
-            if tapeidx + 1 > tapelen
-                debug && reallocatetape()
-                oldtapes = tapes
-                newtapelen = ceil(Int64, tapelen * 1.5)
-                newtapes = Vector{UInt64}[Mmap.mmap(Vector{UInt64}, newtapelen) for i = 1:ncols]
+            # if our initial row estimate was too few, we need to reallocate our tapes/poslens to read the rest of the file
```

Comment:
This should be a much smarter algorithm for re-allocating column tapes if we've underestimated the # of rows to parse. Before, we were just blindly doing `tapelen * 1.5`, which could be disastrous on really large files. We now look at how many bytes are left in the file, how many rows we've parsed vs. how many rows we've parsed, and make a smarter guess at how many rows are left in the file. We then allocate the new tapes one at a time, freeing the previous column's old tape before allocating the next column's new tape.